aci/resource_aci_vnsredirecthealthgroup.go

```diff
@@ -0,0 +1,157 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ciscoecosystem/aci-go-client/client"
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAciL4L7RedirectHealthGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAciL4L7RedirectHealthGroupCreate,
+		ReadContext:   resourceAciL4L7RedirectHealthGroupRead,
+		DeleteContext: resourceAciL4L7RedirectHealthGroupDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAciL4L7RedirectHealthGroupImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"tenant_dn": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		})),
+	}
+}
+
+func getRemoteL4L7RedirectHealthGroup(client *client.Client, dn string) (*models.L4L7RedirectHealthGroup, error) {
+	vnsRedirectHealthGroupCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	vnsRedirectHealthGroup := models.L4L7RedirectHealthGroupFromContainer(vnsRedirectHealthGroupCont)
+	if vnsRedirectHealthGroup.DistinguishedName == "" {
+		return nil, fmt.Errorf("L4 L7Redirect Health Group %s not found", dn)
+	}
+	return vnsRedirectHealthGroup, nil
+}
+
+func setL4L7RedirectHealthGroupAttributes(vnsRedirectHealthGroup *models.L4L7RedirectHealthGroup, d *schema.ResourceData) (*schema.ResourceData, error) {
+	d.SetId(vnsRedirectHealthGroup.DistinguishedName)
+	d.Set("description", vnsRedirectHealthGroup.Description)
+	vnsRedirectHealthGroupMap, err := vnsRedirectHealthGroup.ToMap()
+	if err != nil {
+		return d, err
+	}
+	dn := d.Id()
+	if dn != vnsRedirectHealthGroup.DistinguishedName {
+		d.Set("tenant_dn", "")
+	} else {
+		d.Set("tenant_dn", strings.Split(GetParentDn(vnsRedirectHealthGroup.DistinguishedName, fmt.Sprintf("/"+models.RnvnsRedirectHealthGroup, vnsRedirectHealthGroupMap["name"])), "/svcCont")[0])
```

Comment:
if you are splitting on /svcCont, you probably don't need the GetParentDn. You can either use split or use GetParentDn twice.